playwright-tests/tests/programmes-navigation.spec.ts

```diff
@@ -0,0 +1,66 @@
+import { expect } from '@playwright/test';
+
+import {
+  programmesTestData,
+  programmesExpectedItems,
+} from '@utils/datafactory/programmes.tests';
+import { test } from '@utils/fixtures';
+
+test.describe('Programmes Dropdown Navigation', () => {
+  test('should navigate to all programmes dropdown items', async ({
+    page,
+    basePage,
+  }) => {
+    await basePage.navigateToPath('/');
+
+    for (const item of programmesTestData) {
+      await test.step(`Test navigation to ${item.name}`, async () => {
+        await basePage.navigateToPath('/');
+
+        await basePage.programmesDropdown.click();
+
+        await page.waitForSelector('[data-testid="subNav"]');
+
+        const menuItem = basePage.menuitem(item.name);
+        await menuItem.click();
+
+        await basePage.verifyURL(item.path);
+
+        if (item.expectedText) {
+          await basePage.verifyPageContainsText(item.expectedText);
+        }
+      });
+    }
+  });
+
+  test('should verify programmes dropdown menu items are visible', async ({
+    page,
+    basePage,
+  }) => {
+    await basePage.navigateToPath('/');
+
+    await basePage.programmesDropdown.click();
+
+    await page.waitForSelector('[data-testid="subNav"]');
```

Comment:
I don't think you have to explicitly wait for the dropdown to be opened, since `.toBeVisible()` method is already having auto-waiting mechanism (it will retry till the moment the menuItem will be visible). 
Link to the documentation: https://playwright.dev/docs/test-assertions#auto-retrying-assertions